staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/roundtrip_test.go

```diff
@@ -260,40 +257,50 @@ func TestRoundtrip(t *testing.T) {
 			}{},
 		},
 	} {
-		mps := tc.modePairs
-		if len(mps) == 0 {
+		modePairs := tc.modePairs
+		if len(modePairs) == 0 {
 			// Default is all modes to all modes.
-			mps = []modePair{}
-			for _, em := range allEncModes {
-				for _, dm := range allDecModes {
-					mps = append(mps, modePair{enc: em, dec: dm})
+			modePairs = []modePair{}
+			for _, encMode := range allEncModes {
+				for _, decMode := range allDecModes {
+					modePairs = append(modePairs, modePair{enc: encMode, dec: decMode})
 				}
 			}
 		}
 
-		for _, mp := range mps {
-			encModeName, ok := encModeNames[mp.enc]
+		for _, modePair := range modePairs {
+			encModeName, ok := encModeNames[modePair.enc]
 			if !ok {
 				t.Fatal("test case configured to run against unrecognized encode mode")
 			}
 
-			decModeName, ok := decModeNames[mp.dec]
+			decModeName, ok := decModeNames[modePair.dec]
 			if !ok {
 				t.Fatal("test case configured to run against unrecognized decode mode")
 			}
 
 			t.Run(fmt.Sprintf("enc=%s/dec=%s/%s", encModeName, decModeName, tc.name), func(t *testing.T) {
 				original := tc.obj
 
-				b, err := mp.enc.Marshal(original)
+				b, err := modePair.enc.Marshal(original)
```

Comment:
Please *also* check that these bytes are unmarshalable into the final.Interface().

I think we currently have

1. typedObject to typed-bytes
2. typed-bytes to interface
3. interface to untyped-bytes
4. untyped-bytes to typedObject

this is good, but I'd also like
5. typed-bytes to typedObject

Reply:
Good catch, added that and "untyped-bytes to interface".